File: Python/Searches/linear_search.py
#!/usr/bin/env python3

def main():
    array = []
    size = input("How many numbers do you want to add in array\n")
    while size != 0:
        user_input = int(input())
        array = array + [user_input]
        size = size - 1
    print (array)
    index = linear_search(array, int(input("Enter number to search\n")))
    print(index)

def linear_search(array, num):
    pos  = 0
    for x in array:
        if x == num:
            return(pos)
        pos = pos + 1
    return("Element not found")

if __name__ == '__main__':
    main()

File: C/Searches/linear_search.c
#include<stdio.h>
// Linear search function 
int linear_search(int arr[], int n, int x) 
{ 
    int i;
    for (i = 0; i < n; i++)
        if (arr[i] == x)
         return i;
    return -1;
} 

void main(){
	int n,i,x,choice;
	printf("Enter the number of elements in the array ");
	scanf("%d",&n);
	int a[n];
	printf("Enter the elements: \n");
	for(i=0;i<n;i++){
		scanf("%d",&a[i]);
	}
	printf("Enter the element to be searched ");
	scanf("%d",&x);
	if(linear_search(a,n,x)==-1){
		printf("Element not found\n");
	}
	else{
		printf("Element x = %d is present at %d\n",x,linear_search(a,n,x)+1);
	}
	
	
}
